scrape/target.go

```diff
@@ -75,11 +75,18 @@ func (t *Target) String() string {
 	return t.URL().String()
 }
 
+// MetricMetadataStore represents a storage for metadata.
 type MetricMetadataStore interface {
 	ListMetadata() []MetricMetadata
 	GetMetadata(metric string) (MetricMetadata, bool)
 }
 
+// MetricMetadataAppender provides metadata appends by target against a storage.
+type MetricMetadataAppender interface {
```

Comment:
https://github.com/gotjosh/prometheus/blob/2b2eb79e8b7245d2d60001ae08443d31f7b3a5b9/storage/interface.go#L90-L99

This interface probably belongs in the same place as the storage one, but the small problem here is that both `Target` and `MetricMetadata` belong to the `scrape` package which is not imported as part of storage - and it feels it shouldn't be. Targets we could probably get by just using `labels.Labels` of the target but I don't see a way for `MetricMetadata`, am I missing something?

The trade-off I see is the allocation of new structs (which would belong to the storage package) vs having this interface in a bit of an odd place - I took the latter. Another approach would be an interface for `MetricMetadata` but that seems too complex for this (and not a pattern I've seen used in the codebase)

Reply:
I think it makes sense to have this interface live alongside the existing `Appender` interface. There's already usage of `scrape.Target` by the web code, you could do the same, or consider moving target related data into a new package? Perhaps we should have a metadata package?

Reply:
> Perhaps we should have a metadata package?
I think this is too big of a change to be included as part of this. I'll keep in mind to suggest it as a follow-up.

I'll move the interface to live alongside the other one.